subsys/spm/secure_services.c

```diff
@@ -47,6 +53,42 @@ int spm_secure_services_init(void)
 }
 
 
+#ifdef CONFIG_SPM_SERVICE_READ
+typedef struct {
+	u32_t start;
+	size_t size;
+} read_range_t;
+
+__TZ_NONSECURE_ENTRY_FUNC
+int spm_request_read(void *destination, const void *source, size_t num)
+{
+	static const read_range_t read_ranges[] = {
+#ifdef PM_MCUBOOT_ADDRESS
+		/* Allow reads of mcuboot metadata */
+		{.start = PM_MCUBOOT_PAD_ADDRESS,
+		 .size = PM_MCUBOOT_PAD_SIZE},
+#endif
+	};
+
+	if (destination == NULL || num <= 0) {
+		return -EINVAL;
+	}
+
+	for (int i = 0; i < sizeof(read_ranges)/sizeof(read_range_t); i++) {
```

Comment:
Please use `ARRAY_SIZE`. Also, I use `size_t` when iterating over arrays, for portability; but it doesn't really matter.

Reply:
Should be fixed now.